.github/actions/next-stats-action/src/prepare/repo-setup.js

```diff
@@ -139,41 +163,51 @@ module.exports = (actionInfo) => {
       // wait to pack packages until after dependency paths have been updated
       // to the correct versions
       await Promise.all(
-        Array.from(pkgDatas.keys()).map(async (pkgName) => {
-          const { pkgPath, packedPkgPath } = pkgDatas.get(pkgName)
-
-          let cleanup = null
-
-          if (pkgName === '@next/swc') {
-            // next-swc uses a gitignore to prevent the committing of native builds but it doesn't
-            // use files in package.json because it publishes to individual packages based on architecture.
-            // When we used yarn to pack these packages the gitignore was ignored so the native builds were packed
-            // however npm does respect gitignore when packing so we need to remove it in this specific case
-            // to ensure the native builds are packed for use in gh actions and related scripts
-            await fs.rename(
-              path.join(pkgPath, 'native/.gitignore'),
-              path.join(pkgPath, 'disabled-native-gitignore')
-            )
-            cleanup = async () => {
-              await fs.rename(
-                path.join(pkgPath, 'disabled-native-gitignore'),
-                path.join(pkgPath, 'native/.gitignore')
+        Array.from(pkgDatas.entries()).map(
+          async ([pkgName, { pkgPath, packedPkgPath }]) => {
+            /** @type {null | () => Promise<void>} */
+            let cleanup = null
+
+            if (pkgName === '@next/swc') {
+              // next-swc uses a gitignore to prevent the committing of native builds but it doesn't
+              // use files in package.json because it publishes to individual packages based on architecture.
+              // When we used yarn to pack these packages the gitignore was ignored so the native builds were packed
+              // however npm does respect gitignore when packing so we need to remove it in this specific case
+              // to ensure the native builds are packed for use in gh actions and related scripts
+
+              const nativeGitignorePath = path.join(
+                pkgPath,
+                'native/.gitignore'
+              )
+              const renamedGitignorePath = path.join(
+                pkgPath,
+                'disabled-native-gitignore'
               )
+
+              await fsp.rename(nativeGitignorePath, renamedGitignorePath)
+              cleanup = async () => {
+                await fsp.rename(renamedGitignorePath, nativeGitignorePath)
+              }
             }
-          }
 
-          const { stdout } = await execa('npm', ['pack'], {
-            cwd: pkgPath,
-            env: {
-              ...process.env,
-              COREPACK_ENABLE_STRICT: '0',
-            },
-          })
-          await fs.rename(path.resolve(pkgPath, stdout.trim()), packedPkgPath)
-          if (cleanup) {
-            await cleanup()
+            const { stdout } = await execa(
+              usePnpmPack ? 'pnpm' : 'npm',
```

Comment:
any reason why we wouldn't want to use pnpm all the time?

Reply:
I notice that not every GitHub workflow file has `- run: corepack enable`, so pnpm might not always be available.

`linkPackages` is also used in a lot of places, and I thought it would be safer if `pnpm pack` is a explicitly opt-in.

<img width="510" alt="image" src="https://github.com/vercel/next.js/assets/40715044/952d8b08-4949-4a6e-b8f7-8f148f1ddf6d">

Reply:
> can we try to change to pnpm for all case? as pnpm is the specified pkg manager

Though pnpm is specified in `packageManaer`, We don't have `- run: corepack enable` in every GitHub workflow.

Reply:
yeah we can tackle that later but for this PR I think the purpose was to fully leverage pnpm, as the `corepack enable` was try to make sure pnpm is enabled and we also setup the CI env with pnpm

Reply:
In [`6448770` (#56525)](https://github.com/vercel/next.js/pull/56525/commits/64487705a94838fea22f7599aa0ae9b4f0c0642e) I add "missing" `corepack enable` after every `actions/setup-node` to ensure pnpm is installed.